test/google/storage.go

```diff
@@ -0,0 +1,57 @@
+package google
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"os"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+)
+
+type FakeStorageServer struct {
+	routes map[string]http.HandlerFunc
+}
+
+func (s *FakeStorageServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	for path, handler := range s.routes {
+		if path == r.RequestURI {
+			handler(w, r)
+			return
+		}
+	}
+	w.WriteHeader(http.StatusNotFound)
+}
+
+func NewFakeStorageServer(routes map[string]http.HandlerFunc) (*storage.Client, error) {
+	return newStorageClient(&FakeStorageServer{routes: routes})
+}
+
+func newStorageClient(fakeServer *FakeStorageServer) (*storage.Client, error) {
+	ctx := context.Background()
+	l, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		return nil, err
+	}
+	fakeServerAddr := l.Addr().String()
+	go func() {
+		if err := http.Serve(l, fakeServer); err != nil {
+			panic(err)
+		}
+	}()
+
+	// This prevents the client to use HTTPS
+	_ = os.Setenv("STORAGE_EMULATOR_HOST", fakeServerAddr)
+
+	// Create a client.
+	client, err := storage.NewClient(ctx,
+		option.WithoutAuthentication(),
+		option.WithGRPCDialOption(grpc.WithInsecure()),
+	)
```

Comment:
Can you have a look here https://github.com/googleapis/google-api-go-client/blob/main/testing.md#testing-http-services-using-fakes.

I do not think that we need `option.WithGRPCDialOption` or `STORAGE_EMULATOR_HOST` things here.

Reply:
FYI @sundowndev I removed all options passer to `NewClient`.
We only need to set a proper fake host in `STORAGE_EMULATOR_HOST`

The storage client is very inconsistent, it remain in the gRPC namespace of GCP but under the hoods it uses an http client 🤔